src/H5Dscatgath.c

```diff
@@ -726,6 +730,559 @@ H5D__scatgath_write(const H5D_io_info_t *io_info, const H5D_dset_io_info_t *dset
+                             ((const uint8_t *)write_bufs[i] - (const uint8_t *)io_info->tconv_buf));
+
+                /* Do the data transform before the type conversion (since
+                 * transforms must be done in the memory type). */
+                if (!dset_info->type_info.is_xform_noop) {
+                    H5Z_data_xform_t *data_transform; /* Data transform info */
+
+                    /* Retrieve info from API context */
+                    if (H5CX_get_data_transform(&data_transform) < 0)
+                        HGOTO_ERROR(H5E_DATASET, H5E_CANTGET, FAIL, "can't get data transform info")
+
+                    if (H5Z_xform_eval(data_transform, tmp_write_buf,
+                                       (size_t)io_info->sel_pieces[i]->piece_points,
+                                       dset_info->type_info.mem_type) < 0)
+                        HGOTO_ERROR(H5E_DATASET, H5E_BADVALUE, FAIL, "Error performing data transform")
+                }
+
+                /*
+                 * Perform datatype conversion.
+                 */
+                HDassert(j < bkg_pieces);
+                if (H5T_convert(dset_info->type_info.tpath, dset_info->type_info.src_type_id,
+                                dset_info->type_info.dst_type_id,
+                                (size_t)io_info->sel_pieces[i]->piece_points, (size_t)0, (size_t)0,
+                                tmp_write_buf, bkg_bufs[j]) < 0)
+                    HGOTO_ERROR(H5E_DATASET, H5E_CANTCONVERT, FAIL, "datatype conversion failed")
+
+                /* Advance to next background buffer */
+                j++;
+            }
+        }
+
+        HDassert(j == bkg_pieces);
+    }
+
+    /* Write data to disk */
+    H5_CHECK_OVERFLOW(io_info->pieces_added, size_t, uint32_t)
+    if (H5F_shared_select_write(io_info->f_sh, H5FD_MEM_DRAW, (uint32_t)io_info->pieces_added,
+                                write_mem_spaces, io_info->file_spaces, io_info->addrs,
+                                io_info->element_sizes, write_bufs) < 0)
+        HGOTO_ERROR(H5E_DATASET, H5E_WRITEERROR, FAIL, "selection write failed")
+
+done:
+    /* Release and free selection iterator */
+    if (mem_iter_init && H5S_SELECT_ITER_RELEASE(mem_iter) < 0)
+        HDONE_ERROR(H5E_DATASET, H5E_CANTFREE, FAIL, "Can't release selection iterator")
+    if (mem_iter)
+        mem_iter = H5FL_FREE(H5S_sel_iter_t, mem_iter);
+
+    /* Free write_bufs */
+    H5MM_free(write_bufs);
+    write_bufs = NULL;
+
+    /* Clear and free write_mem_spaces */
+    if (write_mem_spaces) {
+        for (i = 0; i < spaces_added; i++) {
+            HDassert(write_mem_spaces[i]);
+            if (write_mem_spaces[i] != io_info->mem_spaces[i] && H5S_close(write_mem_spaces[i]) < 0)
+                HDONE_ERROR(H5E_DATASET, H5E_CLOSEERROR, FAIL, "Can't close dataspace")
+        }
+        H5MM_free(write_mem_spaces);
+        write_mem_spaces = NULL;
+    }
+
+    /* Free bakcground buffer parameter arrays */
```

Comment:
background

I wonder why codespell is missing a lot of things

Reply:
I've noticed that it's flaky. Not clear why it misses things.